packages/account/src/Sections/Verification/ProofOfAddress/proof-of-address-form.tsx

```diff
@@ -0,0 +1,401 @@
+        };
+
+        const showForm = (should_show_form: boolean) => {
+            setFormState({ ...form_state, ...{ should_show_form } });
+        };
+
+        const onSubmitValues = async (
+            values: TFormInitialValues,
+            { setStatus, setSubmitting }: FormikHelpers<TFormInitialValues>
+        ) => {
+            setStatus({ msg: '' });
+            setFormState({ ...form_state, ...{ is_btn_loading: true } });
+            const settings_values = { ...values };
+            if (values.address_state && states_list.length) {
+                settings_values.address_state = getLocation(states_list, values.address_state, 'value') || '';
+            }
+
+            const data = await WS.setSettings(settings_values);
+
+            if (data.error) {
+                setStatus({ msg: data.error.message });
+                setFormState({ ...form_state, ...{ is_btn_loading: false } });
+                setSubmitting(false);
+                return;
+            }
+
+            const get_settings = await WS.authorized.storage.getSettings();
+
+            if (get_settings.error) {
+                setAPIInitialLoadError(get_settings.error.message);
+                setSubmitting(false);
+                return;
+            }
+            const { address_line_1, address_line_2, address_city, address_state, address_postcode } = get_settings;
+
+            setFormValues({
+                address_line_1,
+                address_line_2,
+                address_city,
+                address_state,
+                address_postcode,
+            });
+
+            // upload files
+            try {
+                const api_response = await file_uploader_ref?.current?.upload();
+                if (api_response?.warning) {
+                    setStatus({ msg: api_response?.message });
+                    setFormState({ ...form_state, ...{ is_btn_loading: false } });
+                    return;
+                }
+
+                const get_account_status = WS.authorized.storage.getAccountStatus();
+
+                if (get_account_status.error) {
+                    setAPIInitialLoadError(get_account_status.error.message);
+                    setSubmitting(false);
+                    return;
+                }
+
+                setFormState({
+                    ...form_state,
+                    ...{ is_submit_success: true, is_btn_loading: false },
+                });
+
+                const { identity, needs_verification } = get_account_status.authentication;
+                const has_poi = !(identity && identity.status === 'none');
+                const needs_poi = needs_verification.length && needs_verification.includes('identity');
+                onSubmit?.(has_poi);
+
+                ['authenticate', 'needs_poa', 'poa_expired'].forEach(key => {
+                    removeNotificationMessage({ key });
+                    removeNotificationByKey({ key });
+                });
```

Comment:
Clarification: Since account_status is fetched before this,Instead of removing notification by key, will refreshNotifications work here?

Reply:
I didn’t check it actually, I left the logic as it was before. 
Basically the flow remains the same. Anyway whole notifications flow should be refactored.
and I believe just exactly for this step it is ok just to remove notification, as user submitted the form. once he or she will enter the app again, or reload the page notifications will be updated